tests/unit/plugins/test_ant_plugin.py

```diff
@@ -0,0 +1,158 @@
+        part_name="my-part", env=f"PATH={str(ant.parent)}", properties=properties
+    )
+    with pytest.raises(errors.PluginEnvironmentValidationError) as raised:
+        validator.validate_environment()
+
+    assert raised.value.reason == "invalid ant version ''"
+
+
+def test_validate_environment_with_ant_part(part_info):
+    properties = AntPlugin.properties_class.unmarshal({"source": "."})
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    validator = plugin.validator_class(
+        part_name="my-part", env="PATH=/foo", properties=properties
+    )
+    validator.validate_environment(part_dependencies=["ant-deps"])
+
+
+def test_validate_environment_without_ant_part(part_info):
+    properties = AntPlugin.properties_class.unmarshal({"source": "."})
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    validator = plugin.validator_class(
+        part_name="my-part", env="PATH=/foo", properties=properties
+    )
+    with pytest.raises(errors.PluginEnvironmentValidationError) as raised:
+        validator.validate_environment(part_dependencies=[])
+
+    assert raised.value.reason == (
+        "'ant' not found and part 'my-part' does not depend on a part named "
+        "'ant-deps' that would satisfy the dependency"
+    )
+
+
+def test_get_build_snaps_and_packages(part_info):
+    properties = AntPlugin.properties_class.unmarshal({"source": "."})
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    assert plugin.get_build_snaps() == set()
+    assert plugin.get_build_packages() == set()
+
+
+def test_get_build_environment(part_info):
+    properties = AntPlugin.properties_class.unmarshal({"source": "."})
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    assert plugin.get_build_environment() == {}
+
+
+def test_missing_parameters():
+    with pytest.raises(ValidationError) as raised:
+        AntPlugin.properties_class.unmarshal({})
+    err = raised.value.errors()
+    assert len(err) == 1
+    assert err[0]["loc"] == ("source",)
+    assert err[0]["type"] == "value_error.missing"
+
+
+def test_get_build_commands(part_info):
+    properties = AntPlugin.properties_class.unmarshal({"source": "."})
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    assert plugin.get_build_commands() == ["ant"] + get_java_post_build_commands()
+
+
+def test_get_build_commands_with_parameters(part_info):
+    properties = AntPlugin.properties_class.unmarshal(
+        {
+            "source": ".",
+            "ant-build-targets": ["compile", "jar"],
+            "ant-build-file": "myfile.txt",
+            "ant-properties": {"prop1": "1", "prop2": "2"},
+        }
+    )
+    plugin = AntPlugin(properties=properties, part_info=part_info)
+
+    assert (
+        plugin.get_build_commands()
+        == ["ant -f myfile.txt -Dprop1=1 -Dprop2=2 compile jar"]
+        + get_java_post_build_commands()
+    )
```

Comment:
Just nitpick and really important, but black likes to format this kind of line in two different ways and the other formatting is what we generally use in craft-parts, I think it's easier to read.
```
assert plugin.get_build_commands() == (
    something
)
```